lib/puppet/provider/selinux_fcontext/semanage.rb

```diff
@@ -71,11 +71,17 @@ def self.instances
     # should never conflict with system policy
     # Old semanage fails with --locallist, use -C
     local_fcs = Selinux.selinux_file_context_local_path
+    fcs = Selinux.selinux_file_context_path
+
+    fcontext_lines = []
+
+    fcontext_lines = parse_fcontext_lines(File.readlines(fcs)) if File.exist? fcs
```

Comment:
I'm not a fan of this guard clause here. It's really hard to read for me. That it was already defined 2 lines above made me think that was redundant. Personally I'd prefer a classic if/else or a ternary.

Given the duplication, how about a helper:

```ruby
def parse_fcontext_file(path)
    return [] unless File.exist?(path)
    parse_fcontext_lines(File.readlines(path))
end

def self.instances
    parse_fcontext_file(Selinux.selinux_file_context_path) + parse_fcontext_file(Selinux.selinux_file_context_local_path)
end
```